Some useful changes for packaging

M0Rf30 opened the pull request:
- Added binary packages generation for
  - Arch Linux
  - Fedora 31
  - openSUSE Tumbleweed
  - Ubuntu 19.10
- Added some details on prebuilt binaries usage
- Travis use latest Ubuntu Bionic agent
- Travis now upload artifacts and sha256sums 
- Travis now build new .pkg.tar.zst Arch Linux packages (+1300% in decompression and installation phase). New extension will be available on next commits.
- Gradle binary is not downloaded during Arch Linux packaging, cause it's in official repos

M0Rf30 committed:
First attempt to build packages

M0Rf30 committed:
Add archlinux target

M0Rf30 committed:
Generate artifacts in pacur/pkgs

M0Rf30 committed:
Deploy artifacts...

M0Rf30 committed:
Add (unneeded) unzip makedep

M0Rf30 committed:
Setup travis to build and deploy packages

M0Rf30 committed:
Support for Arch .tar.zst extension

M0Rf30 committed:
Updated README.md

Polpetta commented on .travis.yml:
```diff
@@ -1,13 +1,41 @@
+  on:
+    repo: Polpetta/jkk
+    branch: master
+    tags: false
```
I'd push binaries only when we create tags, so that a tag corresponds to an official release!

That said, I'd change `prelease` from `true` for `false`. If the user really wants to try the latest version he can install AUR and use the `jkk-git` package :wink: :joy:

M0Rf30 replied:
Done. Waiting for a sparkling release tag :dancer:

Polpetta reviewed (commented):
Everything looks good, I commented a single thing that I'm not sure of, but after that we can merge it :+1:

Polpetta commented:
This PR closes #17 once merged btw

M0Rf30 committed:
Change prerelease to false, waiting for tags mechanism

M0Rf30 committed:
Gradle is downloaded only from repos on Arch

Polpetta committed:
Change releases to be published only when a new tag is created

Polpetta reviewed (approved):
LGTM :+1:

Polpetta committed:
Use new keyword for skipping clean up in Travis

`skip_cleanup` is deprecated, `cleanup` is the new keyword. Additional
information can be gathered here:
https://blog.travis-ci.com/2019-08-27-deployment-tooling-dpl-v2-preview-release

Polpetta commented:
@M0Rf30 the latest changes to the docker image with pacur broke something :confused:  Perhaps we should use tags to be sure to use a stable version every time

M0Rf30 commented:
@Polpetta yes I agree with you. I'm pushing v0.5 tag. This PR can wait

M0Rf30 committed:
Use 0.5 tagged pacur image

M0Rf30 committed:
Add os keyword and replaced aliased api_keys one
